reflex/model.py

```diff
@@ -61,12 +61,11 @@ def __init_subclass__(cls):
         """Drop the default primary key field if any primary key field is defined."""
         non_default_primary_key_fields = [
             field_name
-            for field_name, field in cls.__fields__.items()
-            if field_name != "id"
-            and getattr(field.field_info, "primary_key", None) is True
+            for field_name, field in cls.model_fields.items()
+            if field_name != "id" and getattr(field, "primary_key", None) is True
         ]
         if non_default_primary_key_fields:
-            cls.__fields__.pop("id", None)
+            cls.model_fields.pop("id", None)
```

Comment:
do we need to rebuild the model here?

Reply:
Good catch, I think we always need to rebuild when modifying `model_fields`